op-node/l2/source.go

```diff
@@ -6,29 +6,30 @@ import (
 	"math/big"
 	"time"
 
+	"github.com/ethereum-optimism/optimism/op-node/client"
+
 	"github.com/ethereum/go-ethereum"
 
 	"github.com/ethereum-optimism/optimism/op-node/eth"
 	"github.com/ethereum-optimism/optimism/op-node/rollup"
 	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/rpc"
 )
 
 type Source struct {
-	rpc     *rpc.Client       // raw RPC client. Used for the consensus namespace
-	client  *ethclient.Client // go-ethereum's wrapper around the rpc client for the eth namespace
+	rpc     client.RPC    // raw RPC client. Used for the consensus namespace
+	client  client.Client // go-ethereum's wrapper around the rpc client for the eth namespace
```

Comment:
Nice work on supporting metrics for the raw RPC calls too.

It's in my backlog to not use the `client.Client` bindings here at all; the default bindings have drawbacks (we don't need to decode txs, we're encoding them again after receiving them...) and similarly we're decoding/requesting other things that we don't actually read.

Reply:
And we can keep an eye on the metrics - I suspect we could really use caching well here (like we already have in the L1 source).